synapse/handlers/room_summary.py

```diff
@@ -958,9 +660,8 @@ async def _is_remote_room_accessible(
         ):
             return True
 
-        # Check if the user is a member of any of the allowed spaces
-        # from the response.
-        allowed_rooms = room.get("allowed_room_ids") or room.get("allowed_spaces")
```

Comment:
I wonder if we should leave this longer since I'm pretty sure a response over federation from an older server will still have this?

Reply:
Looks like this impacts Synapse 1.35.0 to 1.40.0, inclusive?
Personally I'd prefer to get the removal over and done with. `allowed_spaces` does come from an _unstable_ implementation after all.

Reply:
That seems reasonable to me, yes.